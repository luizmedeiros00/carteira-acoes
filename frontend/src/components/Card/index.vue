<template>
  <div class="block p-3 bg-white rounded-lg border border-gray-200 w-full shadow-md">
    <h2 v-if="title" class="text-base font-medium text-slate-500 mb-5">{{ title }}</h2>
    <slot></slot>
  </div>
</template>

<script setup lang="ts">

  interface Props {
    title?: string
  }
  const { title } = defineProps<Props>()

</script>
