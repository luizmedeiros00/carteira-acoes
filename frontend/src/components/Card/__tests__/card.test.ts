import { shallowMount, mount, VueWrapper } from '@vue/test-utils'
import Card from '../index.vue'

describe('Card component', () => {
  type Instance = InstanceType<typeof Card>
  let mountFunction: (options?: object) => VueWrapper<Instance>

  beforeEach(() => {
    mountFunction = (options = {}) => {
      return shallowMount(Card, {
        ...options,
      })
    }
  })

  test('should render component with props title and snapshot', () => {
    const wrapper = mountFunction({
      props: {
        title: 'Title Card'
      }
    })
    expect(wrapper.find('h2').exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render component without props title and snapshot', () => {
    const wrapper = mountFunction({})

    expect(wrapper.find('h2').exists()).toBe(false)
    expect(wrapper.html()).toMatchSnapshot()
  })
})