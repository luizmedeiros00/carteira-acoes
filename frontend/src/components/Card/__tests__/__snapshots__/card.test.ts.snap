// Vitest Snapshot v1

exports[`Card component > should render component with props title and snapshot 1`] = `
"<div class=\\"block p-3 bg-white rounded-lg border border-gray-200 w-full shadow-md\\">
  <h2 class=\\"text-base font-medium text-slate-500 mb-5\\">Title Card</h2>
</div>"
`;

exports[`Card component > should render component without props title and snapshot 1`] = `
"<div class=\\"block p-3 bg-white rounded-lg border border-gray-200 w-full shadow-md\\">
  <!--v-if-->
</div>"
`;
